sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -226,7 +228,8 @@ boolean isConnectionSharingAcrossClientsEnabled() {
      *
      * @return the token resolver
      */
-    CosmosAuthorizationTokenResolver getAuthorizationTokenResolver() {
+    @Beta(value = Beta.SinceVersion.V4_24_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public CosmosAuthorizationTokenResolver getAuthorizationTokenResolver() {
```

Comment:
none of the other APIs in the CosmosClientBuilder have a public getter.

Can we avoid exposing getter public api in the CosmosClientBuilder to be consistent with the other APIs in the CosmosClientBuilder?

Reply:
I removed it for now, though there's a special case to be made for CosmosAuthorizationTokenResolver, that is if the user wants to update its output on demand based on catching a 401/403 Cosmos exception, they could do that through the Cosmos client rather than keeping a separate instance of it.
We might have to add this back depending on the specific feedback and use case for which we are adding this.